### ISSUE TITLE
Mocks do not support embedded interfaces

### ISSUE THREAD

**Issue by inconshreveable** [sourcegraph/gen-mocks#5]
Asking gen-mocks to generate mocks for `type A` will fail because the code which walks the field list of an interface only handles function types and not embedded interfaces:

``` go
type B interface {
    DoB()
}

type A interface {
    DoA()
    B
}
```

**Comment by beyang** [comment]
Thanks for filing the issue -- will hopefully take a look soon but in the meantime, PRs welcome :)

**Comment by inconshreveable** [comment]
Yeah, took a look at it and basically in order to do this you'd have to actually do types analysis with `go/types` so it would require a large refactoring and change in scope of the program, may not be worth it tbh. In this circumstance I've switched to using mockery/testify which does the types analysis.

(was using this to mock out grpc services which in the latest revisions has started using embedded interfaces)

**Comment by beyang** [comment]
Got it, mockery looks awesome, gonna check that out!
